packages/core/src/new-api/internal/validation/futures/validateArtifactContractDeployment.ts

```diff
@@ -1,13 +1,34 @@
 import { ethers } from "ethers";
 
 import { IgnitionValidationError } from "../../../../errors";
+import { isModuleParameterRuntimeValue } from "../../../type-guards";
 import { ArtifactResolver } from "../../../types/artifact";
-import { ArtifactContractDeploymentFuture } from "../../../types/module";
+import {
+  ArtifactContractDeploymentFuture,
+  ModuleParameters,
+} from "../../../types/module";
 
 export async function validateArtifactContractDeployment(
   future: ArtifactContractDeploymentFuture,
-  artifactLoader: ArtifactResolver
+  artifactLoader: ArtifactResolver,
+  moduleParameters: ModuleParameters
 ) {
+  const moduleParams = future.constructorArgs.filter(
```

Comment:
I think this wouldn't validate something like:

```
m.contract("A", artifact, [[m.getParam("p")]])
```

because the param is nested in an array.

I think we'd need to do something similar to what @kanej made when replacing complex args with other representations.

Reply:
This comment is also valid for the rest of the futures that take constructor or call arguments.

Reply:
Here is the `replace-within-arg` code that has to do a recursive breakdown of the argument list @zoeyTM:

https://github.com/NomicFoundation/ignition/blob/1d20a732a93bcc03b2260b78fe76b613ea7bcd6e/packages/core/src/new-api/internal/utils/replace-within-arg.ts#L48

Reply:
fixed in bac1d82